music21/musicxml/m21ToXml.py

```diff
@@ -3472,7 +3521,23 @@ def noteToXml(self, n: note.GeneralNote, noteIndexInChord=0, chordParent=None):
             for t in mxTieList:
                 mxNote.append(t)
 
-        # TODO: instrument
+        # instrument tags are necessary when there is more than one
+        if self.parent is not None and len(self.parent.instrumentStream) > 1 and not n.isRest:
+            if chordParent:
+                closest_inst = chordParent.getContextByClass(Instrument, followDerivation=False)
+                assert closest_inst in self.parent.instrumentStream
+            else:
+                closest_inst = n.getContextByClass(Instrument, followDerivation=False)
+                assert closest_inst in self.parent.instrumentStream
+            instance_to_use = self.parent.instrumentStream[type(closest_inst)].first()
+            if instance_to_use is None:
+                # exempt coverage, because this is only for safety/unreachable
+                raise MusicXMLExportException(f'Could not find {closest_inst} for note {n}'
+                    + 'in instrumentStream')  # pragma: no cover
+            mxInstrument = SubElement(mxNote, 'instrument')
+            assert instance_to_use.instrumentId, id(instance_to_use)
+            mxInstrument.set('id', instance_to_use.instrumentId)
+
```

Comment:
Can this be pulled out into a separate routine and documented a bit more.

also I don't like bare-asserts in code because if it fails then someone just gets AssertionError and a random id with nothing they can do to fix it.